### PR TITLE
Remove merge_params.force_remove_source_branch

### PR DESCRIPTION
**Pull request by philhug**
Fixes failed webhooks if remove source branch is enabled on branch.
Closes: #300

### CODE REVIEW

File: scm/driver/gitlab/webhook.go [hunk 1]
```diff
@@ -601,21 +601,19 @@ type (
...
+			DeletedAt                 interface{} `json:"deleted_at"`
+			Description               string      `json:"description"`
+			HeadPipelineID            interface{} `json:"head_pipeline_id"`
+			ID                        int         `json:"id"`
+			Iid                       int         `json:"iid"`
+			LastEditedAt              interface{} `json:"last_edited_at"`
+			LastEditedByID            interface{} `json:"last_edited_by_id"`
+			MergeCommitSha            string      `json:"merge_commit_sha"`
+			MergeError                interface{} `json:"merge_error"`
+			MergeParams               interface{} `json:"-"`
```
Comment (ankitm123) [inline scm/driver/gitlab/webhook.go:616]: Why is this removed? Should it be changed instead to 
```
ForceRemoveSourceBranch bool `json:"force_remove_source_branch"`
```

and verified that gitlab sends boolean instead of string (also fix testdata)?

**Comment by garethjevans** [comment]
Closing this in favour of #301

File: scm/driver/gitlab/webhook.go (hunk 1 above)
Comment (zzhzero) [reply scm/driver/gitlab/webhook.go:616]: [force_remove_source_branch type is inconsistent](https://gitlab.com/gitlab-org/gitlab/-/issues/15647)
